File: DSA-2019/week-02/day-1/sharpieSet.py
from sharpie import Sharpie


class SharpieSet():
    def __init__(self):
        self.sharpies = []
        self.usable_sharpies = []
        self.usable_sharpies_count = 0

    def add_sharpie(self, sharpie: Sharpie):
        self.sharpies.append(sharpie)

    def count_usable(self):
        for i in self.sharpies:
            if (i.ink_amount > 0):
                self.usable_sharpies.append(i)
                self.usable_sharpies_count += 1

    def remove_unusable(self):
        for i in self.sharpies:
            if (i.ink_amount <= 0):
                self.sharpies.remove(i)

File: DotNet-2018/week2/day2/week2-day2/week2-day2/SharpieSet.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace week2_day2
{
    class SharpieSet
    {
        public List<Sharpie> sharpielist = new List<Sharpie>();
        public List<int> usableList=new List<int>();
        public List<Sharpie> unusableList=new List<Sharpie>();
        public List<int> CountUsable()
        {
            for (int i=0; i<this.sharpielist.Count;i++)
            {
                if (sharpielist[i].inkAmount > 0)
                {
                    usableList.Add(i);
                }
                else
                {
                    unusableList.Add(sharpielist[i]);
                }
            }

            return this.usableList;
        }

        public void Removetrash()
        {
            foreach (Sharpie i in unusableList)
            {
                sharpielist.Remove(i);
            }
        }
        

    }
}
